packages/main/src/plugin/extension-loader.ts

```diff
@@ -725,7 +725,10 @@ export class ExtensionLoader {
       },
       getProviderLifecycleContext(
         providerId: string,
-        providerConnectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo,
+        providerConnectionInfo:
+          | containerDesktopAPI.ContainerProviderConnection
+          | ProviderContainerConnectionInfo
+          | ProviderKubernetesConnectionInfo,
```

Comment:
```suggestion
        providerConnectionInfo: containerDesktopAPI.ContainerProviderConnection
```

IMO there is no need for union type here, because you cannot use internal types from client extension.